descartes_core/include/descartes_core/path_planner_base.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * path_planner_base.h
+ *
+ *  Created on: Jan 19, 2015
+ *      Author: ros developer 
+ */
+
+#ifndef DESCARTES_CORE_PATH_PLANNER_BASE_H_
+#define DESCARTES_CORE_PATH_PLANNER_BASE_H_
+
+#include <descartes_core/trajectory_pt.h>
+#include <descartes_core/robot_model.h>
+#include <vector>
+
+namespace descartes_core
+{
+
+class PathPlannerBase
+{
+public:
+  virtual ~PathPlannerBase(){}
+
+  virtual bool initialize(RobotModelConstPtr &model,const std::vector<TrajectoryPtPtr>& traj,
+            double sampling = 0.1f) = 0;
+
+  virtual bool addAfter(const TrajectoryPt::ID& ref_id,TrajectoryPtPtr cp) = 0;
```

Comment:
These modifiers should be removed from the base class at some point.  However, we should go through the effort of defining their function since modifiers will be required, albeit in another class.